Adding retries option

ken210 opened the pull request:
Hello! First of all i wish to thank you guys for the great project.

Sometimes, when you got any problem with your network, and your upload fails, you can't retry it.
So I added the `retries` option that allows you to set how many times you wish gulp-s3 will retry to send your file.

``` JavaScript
var s3 = require("gulp-s3");
var aws = {
  "key": "AKIAI3Z7CUAFHG53DMJA",
  "secret": "acYxWRu5RRa6CwzQuhdXEfTpbQA+1XQJ7Z1bGTCx",
  "bucket": "dev.example.com",
  "region": "eu-west-1"
};
var options = {
  headers: {
    'Cache-Control': 'max-age=315360000, no-transform, public'
  },
  retries: 3
};
gulp.src('./dist/**')
    .pipe(s3(aws, options));
```

ken210 committed:
Adding retries options like {retries: 3}

ken210 committed:
bumping version

ken210 committed:
typo

cworsley4 commented:
Perhaps this is more of an AWS question, but since S3 doesn't have regions (its "global") should the regions prop be included here?

ken210 committed:
updating es and knox

ken210 committed:
this commit does the following:

- change from es.mapSync to es.map to be async
- add a FILE_TIMEOUT const of 5 min
- add a RETRY_TIMEOUT const of 30 sec witch automacally retry (for unresposive network)
- better error handling
- less verbose output
